src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectConfigurationFileChangeEventArgs.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common.Serialization;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class ProjectConfigurationFileChangeEventArgs
+    {
+        private readonly JsonFileDeserializer _jsonFileDeserializer;
+        private FullProjectSnapshotHandle _projectSnapshotHandle;
+        private object _projectSnapshotHandleLock;
+        private bool _deserialized;
+
+        public ProjectConfigurationFileChangeEventArgs(
+            string configurationFilePath,
+            RazorFileChangeKind kind) : this(configurationFilePath, kind, JsonFileDeserializer.Instance)
+        {
+        }
+
+        // Internal for testing
+        internal ProjectConfigurationFileChangeEventArgs(
+            string configurationFilePath,
+            RazorFileChangeKind kind,
+            JsonFileDeserializer jsonFileDeserializer)
+        {
+            if (configurationFilePath is null)
+            {
+                throw new ArgumentNullException(nameof(configurationFilePath));
+            }
+
+            if (jsonFileDeserializer is null)
+            {
+                throw new ArgumentNullException(nameof(jsonFileDeserializer));
+            }
+
+            ConfigurationFilePath = configurationFilePath;
+            Kind = kind;
+            _jsonFileDeserializer = jsonFileDeserializer;
+            _projectSnapshotHandleLock = new object();
+        }
+
+        public string ConfigurationFilePath { get; }
+
+        public RazorFileChangeKind Kind { get; }
+
+        public bool TryDeserialize(out FullProjectSnapshotHandle handle)
+        {
+            if (Kind == RazorFileChangeKind.Removed)
+            {
+                // There's no file to represent the snapshot handle.
+                handle = null;
+                return false;
```

Comment:
It's never reset because the class is immutable. Basically it defaults to `false` so if the `Kind` is `Removed` this method will never actually deserialize